src/libraries/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -91,5 +122,99 @@ public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, int i
 
             return default;
         }
+
+        /// <summary>Returns the element at a specified index in a sequence or a default value if the index is out of range.</summary>
+        /// <param name="source">An <see cref="IEnumerable{T}" /> to return an element from.</param>
+        /// <param name="index">The index of the element to retrieve, which is either from the start or the end.</param>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source" />.</typeparam>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        /// <returns>
+        ///   <see langword="default" /> if <paramref name="index" /> is outside the bounds of the <paramref name="source" /> sequence; otherwise, the element at the specified position in the <paramref name="source" /> sequence.
+        /// </returns>
+        public static TSource? ElementAtOrDefault<TSource>(this IEnumerable<TSource> source, Index index)
+        {
+            if (source == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
+            }
+
+            if (!index.IsFromEnd)
+            {
+                return source.ElementAtOrDefault(index.Value);
+            }
+
+            TryGetElementAt(source, index.Value, out TSource? result);
+            return result;
+        }
+
+        private static bool TryGetElementAt<TSource>(IEnumerable<TSource> source, int indexFromEnd, [MaybeNullWhen(false)] out TSource? result)
```

Comment:
```suggestion
        private static bool TryGetElementAt<TSource>(IEnumerable<TSource> source, int indexFromEnd, [MaybeNullWhen(false)] out TSource result)
```

Reply:
Are you sure about this? If `result` may be null, it should nullable. In LINQ, we have `TElement? TryGetElementAt(int index, out bool found);` for `IPartition<TElement>`, etc.

Reply:
I'm sure. The attribute is saying that it could be null if the method returns false.  If you were to also have the question mark, that's saying it could also be null if the method returns true,  even if the developer specified a non-nullable TSource.

Reply:
Updated.